### PR TITLE
psi-notify: init at 1.2.1

### PR DESCRIPTION
**Pull request by eduarrrd**
###### Motivation for this change

New package I use.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/psi-notify/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "cdown";
+    repo = pname;
+    rev = version;
+    sha256 = "0hn37plim1smmlrjjmz8kybyms8pz3wxcgf8vmqjrsqi6bfcym7g";
+  };
+
+  buildInputs = [ systemd libnotify pkg-config ];
```
Comment (figsoda) [inline pkgs/applications/misc/psi-notify/default.nix:14]:
```suggestion
  nativeBuildInputs = [ pkg-config ];

  buildInputs = [ libnotify systemd ];
```

File: pkgs/applications/misc/psi-notify/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  buildPhase = ''
+    make
+  '';
```
Comment (figsoda) [inline pkgs/applications/misc/psi-notify/default.nix:17]: this is the default
```suggestion

```

File: pkgs/applications/misc/psi-notify/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  installPhase = ''
+    install -D psi-notify $out/bin/psi-notify
+    substituteInPlace psi-notify.service --replace psi-notify $out/bin/psi-notify
+    install -D psi-notify.service $out/share/systemd/user/psi-notify.service
+  '';
```
Comment (figsoda) [inline pkgs/applications/misc/psi-notify/default.nix:22]:
```suggestion
  installPhase = ''
    runHook preInstall
    install -D psi-notify $out/bin/psi-notify
    substituteInPlace psi-notify.service --replace psi-notify $out/bin/psi-notify
    install -D psi-notify.service $out/share/systemd/user/psi-notify.service
    runHook postInstall
  '';
```

File: pkgs/applications/misc/psi-notify/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    description = "Alert on system resource saturation";
+    longDescription = ''
+      psi-notify can alert you when resources on your machine are becoming
+      oversaturated, and allow you to take action before your system slows to a
+      crawl.
+    '';
+    license = licenses.mit;
+    homepage = "https://github.com/cdown/psi-notify";
+    platforms = platforms.linux;
+    maintainers = [ maintainers.eduarrrd ];
```
Comment (figsoda) [inline pkgs/applications/misc/psi-notify/default.nix:34]:
```suggestion
    maintainers = with maintainers; [ eduarrrd ];
```

**Review by figsoda — CHANGES_REQUESTED** [review]

**Comment by figsoda** [comment]
upstream has a togglable feature `WANT_SD_NOTIFY` can be set to 0 and make the libsystemd dependency optional
https://github.com/cdown/psi-notify/blob/1.2.1/Makefile#L4

**Comment by eduarrrd** [comment]
Addressed the suggestions, they make sense to me.

> upstream has a togglable feature `WANT_SD_NOTIFY` that can be set to 0 and make the libsystemd dependency optional
> https://github.com/cdown/psi-notify/blob/1.2.1/Makefile#L4
> 
> maybe it could be added to this package too?

Possibly, but I don't intend to use it without that. We can discuss this once there's a user who intends to use it this way.

**Comment by figsoda** [comment]
fair enough

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 139335` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>psi-notify</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Pull request merged**